tests/vm_console_proxy_test.go

```diff
@@ -0,0 +1,188 @@
+package tests
+
+import (
+	"reflect"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	ssp "kubevirt.io/ssp-operator/api/v1beta1"
+	vm_console_proxy "kubevirt.io/ssp-operator/internal/operands/vm-console-proxy"
+)
+
+var _ = Describe("VM Console Proxy Operand", func() {
+	var (
+		clusterRoleResource        testResource
+		clusterRoleBindingResource testResource
+		serviceAccountResource     testResource
+		serviceResource            testResource
+		deploymentResource         testResource
+		configMapResource          testResource
+	)
+
+	BeforeEach(func() {
+		strategy.SkipSspUpdateTestsIfNeeded()
+
+		updateSsp(func(foundSsp *ssp.SSP) {
+			if foundSsp.GetAnnotations() == nil {
+				foundSsp.Annotations = make(map[string]string)
+			}
+
+			namespace := strategy.GetVmConsoleProxyNamespace()
```

Comment:
This line panics for update tests. They use the `existingSspStrategy`, and the annotation is not defined at the time that this line is called.

Reply:
Thanks, I see. I'll investigate tomorrow to see where to define that annotation.

Reply:
Isn't `existingSspStrategy` used only when ssp-operator is deployed by HCO? I don't see in the logs any indication that the tests are deployed by HCO, is it done on QE side?

If so, then I think we don't need to panic and just return the default namepsace:
```
	namespaceStr, ok := s.ssp.ObjectMeta.GetAnnotations()[vm_console_proxy.VmConsoleProxyNamespaceAnnotation]
	if !ok {
		return vm_console_proxy.DefaultNamespace // "kubevirt"
	}
	return namespaceStr
```
Am I right?

Reply:
The `existingSspStrategy` should be used when the SSP CR already exists in the cluster. We use it for the update tests. Here is the script run by CI: https://github.com/kubevirt/ssp-operator/blob/be8d6b510301a9658018e6796a702082dbd3eafd/automation/e2e-upgrade-functests/run.sh

It does these steps:
- Install the latest released ssp-operator version
- Create an SSP CR
- Wait until ssp-operator is finished deploying all resources
- Install the ssp-operator build from the current PR
- Run the functional test suite

This way we can test if there are any problems when updating from the latest released version to the PR version.

Reply:
Thanks! I see how it works now, learned something new. So in this PR we should also add new annotations to `ssp-sample.yaml` [1] right?

[1] https://github.com/kubevirt/tekton-tasks-operator/blob/main/scripts/ssp-sample.yaml

Reply:
That file is in TTO, do you mean `config/samples/ssp_v1beta1_ssp.yaml` ? It only contains mandatory parts, so I would say don't add it there.

Can you add code to read the kubevirt namespace for tests from an env variable? This file contains env variables used by tests: https://github.com/kubevirt/ssp-operator/blob/3719aa2cf60a2f93117f02c57a380455fef56bdd/tests/env/env.go
You could add a new one and then define it in the update tests script in `automation/e2e-upgrade-functests/run.sh`

Reply:
@akrejcir Ah no, I took a look on how it was implemented in TTO and here in ssp-operator. I meant to [1] of course, wrong copy-paste.

[1] https://github.com/kubevirt/ssp-operator/blob/master/automation/e2e-upgrade-functests/ssp-cr-template.yaml

Reply:
So `func (s *existingSspStrategy) GetVmConsoleProxyNamespace() string` should read it from env var and not from an annotation?

Reply:
No, in case the annotation is not defined, then it should read env variable.

Reply:
> @akrejcir Ah no, I took a look on how it was implemented in TTO and here in ssp-operator. I meant to [1] of course, wrong copy-paste.
> 
> [1] https://github.com/kubevirt/ssp-operator/blob/master/automation/e2e-upgrade-functests/ssp-cr-template.yaml

Right. But no, don't edit this file. We still need the logic with the env variable for QE tests.

Reply:
Ack, thanks. Done. Thanks a lot for helping here, really appreciated. :)